### ISSUE TITLE
Implement geolocation control instead of prompting for location by default

### ISSUE THREAD

**Issue by caseycesari** [WikiWatershed/model-my-watershed#2540]
When the application is loaded, the user's location is asked for unprompted.

![image](https://user-images.githubusercontent.com/1042475/33442933-50683732-d5c4-11e7-9c67-ae7c4d9e978e.png)

And Chrome outputs a warning in the console:

> [Violation] Only request geolocation information in response to a user gesture.

Here's some more information from Google regarding this: https://developers.google.com/web/fundamentals/native-hardware/user-location/.

After reading that, and visiting some other sites, it does seem like not a great user experience to ask for the user's location unprompted.

We should implement a control (near the geocoder, for instance), that, when clicked will ask for the user's location and, if they comply, move the map to their location.

**Comment by ajrobbins** [comment]
I think it's a better user experience for the app to be centered on the user's location from the start, using the initial location prompt. That way they can go directly to selecting an AoI, especially since the boundary selection doesn't work at a low zoom level aka default zoom level. If the geolocation is tucked in a button, the user will start out seeing the whole country but may not understand why they can't select by boundaries, and will need to zoom and pan more. Thoughts @jfrankl ?

**Comment by caseycesari** [comment]
True. Some sort of prompt, like a modal with a location button, may work.

**Comment by jfrankl** [comment]
What if we prompt for location after the user clicks "Select area"? I think it would be an improvement over our current implementation, but also keep it relatively seamless.

Clicking on "Select area" seems like a reasonable gesture for us to prompt the user asking for their location. I agree with Casey and with Google, that prompting upon initial page load isn't a great experience. Imagine that you don't know what MMW is about. You land on the page for the first time. And before you are even reading about what it does, it's asking you for your location. When a user says they want to select an area, that action is related to location, and I think it will be relatively intuitive why we are asking for their location at that point.

This is also assuming that our users want to zoom into their location—that they are researching the area where they are located. That seems reasonable, but it might be worth looking into it.

**Comment by ajrobbins** [comment]
>What if we prompt for location after the user clicks "Select area"?

I like that idea, and agree with your assessment

**Comment by caseycesari** [comment]
By "Select Area", do you mean "Get Started"?

![image](https://user-images.githubusercontent.com/1042475/33447410-37ea8bda-d5d1-11e7-97ce-d331cc603fc1.png)

**Comment by jfrankl** [comment]
Sorry—yes, I did mean "Get started".
